### PR TITLE
WIP - Forms

### PR DESCRIPTION
**Pull request by kivra-thohut**

### CODE REVIEW

File: swagger.yml [hunk 1]
```diff
@@ -2046,6 +2062,280 @@ paths:
...
+        200:
+          description: |
+            Form response ids
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/FormResponseIds"
+        400:
+          description: |
+            The request is malformed. This happens when the formId is not a UUID (V4).
```
Comment (chsukivra) [inline swagger.yml:2219]: It can probably happen because of other reasons. But it is good to have a likely reason to lead someone figuring out.

**Review by chsukivra — APPROVED** [review]
Nice review session. I think you'll do the changes.

File: swagger.yml [hunk 2]
```diff
@@ -749,6 +750,11 @@ info:
 tags:
   - name: "Tenant API - Content"
     description: Endpoints for matching users and sending content
+  - name: "Tenant API - Forms"
```
Comment (paolokivra) [inline swagger.yml:753]: "Tenant API - Forms (BETA)"

File: swagger.yml [hunk 3]
```diff
@@ -762,6 +768,7 @@ x-tagGroups:
   - name: Tenant API
     tags:
       - "Tenant API - Content"
+      - "Tenant API - Forms"
```
Comment (paolokivra) [inline swagger.yml:771]: "Tenant API - Forms (BETA)

File: swagger.yml [hunk 4]
```diff
@@ -2046,6 +2066,290 @@ paths:
...
                       data: PGgxPmNvbnRlbnQgMTwvaDE+CjxwPmNvbnRlbnQgMTwvcD4=
                       content_type: text/html
+              
+  # ##############################################
+  # POST, GET /v1/tenant/TKEY/form
+  # ##############################################
+  /v1/tenant/{tenantKey}/form:
+    post:
+      tags:
+        - "Tenant API - Forms"
```
Comment (paolokivra) [inline swagger.yml:2076]: "Tenant API - Forms (BETA)

File: swagger.yml [hunk 5]
```diff
@@ -2046,6 +2066,290 @@ paths:
...
+          description: |
+            Not authorized to make the request.
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: "#/components/schemas/FormTemplate"
+    get:
+      tags:
+        - "Tenant API - Forms"
```
Comment (paolokivra) [inline swagger.yml:2118]: "Tenant API - Forms (BETA)

File: swagger.yml [hunk 6]
```diff
@@ -2046,6 +2066,290 @@ paths:
...
+          description: |
+            Not authorized to make the request
+
+  # ##############################################
+  # GET /v1/tenant/{tkey}/form/{formId}
+  # ##############################################
+  /v1/tenant/{tenantKey}/form/{formId}:
+    get:
+      tags:
+        - "Tenant API - Forms"
```
Comment (paolokivra) [inline swagger.yml:2150]: "Tenant API - Forms (BETA)

File: swagger.yml [hunk 7]
```diff
@@ -2046,6 +2066,290 @@ paths:
...
+          description: |
+            Form template not found
+ 
+  # ##############################################
+  # GET /v1/tenant/{tkey}/form/{formid}/responses
+  # ##############################################
+  /v1/tenant/{tenantKey}/form/{formId}/responses:
+    get:
+      tags:
+        - "Tenant API - Forms"
```
Comment (paolokivra) [inline swagger.yml:2196]: "Tenant API - Forms (BETA)

File: swagger.yml [hunk 8]
```diff
@@ -2046,6 +2066,290 @@ paths:
...
+            The form template associated with formId does not exist.
+
+
+  # ##############################################
+  # GET, DELETE /v1/tenant/{tkey}/form/{formid}/responses/{respid}
+  # ##############################################
+  /v1/tenant/{tenantKey}/form/{formId}/responses/{responseId}:
+    get:
+      tags:
+        - "Tenant API - Forms"
```
Comment (paolokivra) [inline swagger.yml:2249]: "Tenant API - Forms (BETA)

File: swagger.yml [hunk 9]
```diff
@@ -2046,6 +2066,290 @@ paths:
...
+            Not authorized to make the request
+        403:
+          description: |
+            Forbidden from accessing the form response ids
+        404:
+          description: |
+            Either the form template associated with the given formId does not exist or the form response associated with the given responseId does not exist.
+    delete:
+      tags:
+        - "Tenant API - Forms"
```
Comment (paolokivra) [inline swagger.yml:2303]: "Tenant API - Forms (BETA)

**Review by paolokivra — COMMENTED** [review]
Just add "(BETA)" to the `tag`

**Pull request merged**